sklearn/utils/estimator_checks.py

```diff
@@ -1443,41 +1443,79 @@ def check_supervised_y_2d(name, estimator_orig):
     assert_allclose(y_pred.ravel(), y_pred_2d.ravel())
 
 
-@ignore_warnings(category=(DeprecationWarning, FutureWarning))
+@ignore_warnings
+def check_classifiers_classes_helper(X, y, name, classifier_orig):
+    classes = np.unique(y)
+    classifier = clone(classifier_orig)
+    if name == 'BernoulliNB':
+        X = X > X.mean()
+    set_random_state(classifier)
+    # fit
+    classifier.fit(X, y)
+
+    y_pred = classifier.predict(X)
+
+    if hasattr(classifier, "decision_function"):
+        decision = classifier.decision_function(X)
+        n_samples, n_features = X.shape
+        if len(classes) == 2:
+            assert_equal(decision.shape, (n_samples, ))
+            dec_pred = (decision.ravel() > 0).astype(np.int)
+            assert_array_equal(classifier.classes_[dec_pred], y_pred)
+        elif (len(classes) == 3 and
```

Comment:
thanks, yes, we don't need `len(classes) == 3` especially if we `getattr(classifier, 'decision_function_shape', 'ovr') == 'ovr')` instead of `isinstance`